code/dqcommon/savedata.q

```diff
@@ -1,16 +1,23 @@
 \d .dqe
-savedata:{[dir;pt;ns;tabname]
+savedata:{[dir;pt;savetemp;ns;tabname]
   .lg.o[`dqe;"Saving ",(string tabname)," data to ",.os.pth dir];
   pth:` sv .Q.par[dir;pt;tabname],`;
   err:{[e].lg.e[`savedata;"Failed to save dqe data to disk : ",e];'e};
-  .[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;ns tabname]]);err];
-  .lg.o[`delete;"deleting ",(string tabname)," data from in-memory table"];                                     /- empty the table from memory
+  tab:.Q.dd[ns;tabname];
+  .[upsert;(pth;.Q.en[dir;r:0!.save.manipulate[tabname;{[tab;x]exec from tab where i=x}[tab]'[savetemp] ]]);err];
+  .lg.o[`savedata;"i values for rows that will be saved down: ","," sv string savetemp];
```

Comment:
How many indexes will be passed in this `savetemp` variable? i.e. will this actually be a useful log message?

Reply:
Could be up to the thousands - the log message will look like
"i values for rows that will be saved down: 188,189,191,190,192,193,195,194,196,197,199,198,200,201,202,204,203,205,206,208,207,209,210,211

which i am unsure if it would be useful